src/EnzymeRules/enzymerules.jl

```diff
@@ -155,7 +155,16 @@ for AMT in (:Matrix, :SparseMatrixCSC)
                 if verbose > 0
                     @info "($psolver, $pamt) reverse"
                 end
-                _b.dval .= Krylov.$solver(transpose(A), bx[]; M=M, N=N, verbose=verbose, options...)[1]
+                N = M
+                M = N
+                if isa(M, IncompleteLU.ILUFactorization)
```

Comment:
You need to do the same thing if `N` is an `ILUFactorization` (right preconditioner).

Reply:
I'll do this in another PR.